src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -444,6 +442,13 @@ CitusModifyBeginScan(CustomScanState *node, EState *estate, int eflags)
 		RebuildQueryStrings(jobQuery, taskList);
 	}
 
+	/*
+	 * Modify tasks are always assigned using first-replica policy. Since tasks
+	 * are also sorted by anchor shard id within this function, we do not need
+	 * to sort them again while obtaining locks.
+	 */
+	taskList = FirstReplicaAssignTaskList(taskList);
```

Comment:
This is too early to assign tasks to placements since we haven't obtained the metadata locks yet.

Reply:
Should I sort them in the planner ?

Reply:
why do you need to sort them?

Reply:
To obtain locks in the correct order within `AcquireMetadataLocks`  (https://github.com/citusdata/citus/pull/2124#issuecomment-384171739)